app/controllers/users_controller.rb

```diff
@@ -1,9 +1,32 @@
 class UsersController < ApplicationController
   def new
-    # @user = User.new
+    @user = User.new
   end
 
   def create
+    # @user = User.new({username: params[:username], email: params[:email], password: params[:password]})
+    @user = User.new(user_params)
 
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+    render :edit
```

Comment:
Yes, I thought about this. They both lead to the same edit action, under both conditions. So, it doesn't make any difference. The actual action, if updated, should redirect to user_path i.e. the show action, but we are not doing any show action in this project. I'm proud to have changed this, just because you wanted me to